### ISSUE TITLE
Crash on Test Input

### ISSUE THREAD

**Issue by jcburley** [jcburley/lisp-zero-go#1]
When reading **Z.lisp**, a cut-down version of **zero-test.lisp**, **lisp-zero-go** crashes in ways that seem to vary by invocation or at least by re-build.

E.g.:

```
(dlv) c
()
()
()
()
()
unexpected fault address 0x0
fatal error: fault
> [unrecovered-panic] runtime.startpanic() /usr/local/go/src/runtime/panic.go:588 (hits goroutine(1):1 total:1) (PC: 0x42c340)
Warning: debugging optimized function
	runtime.curg._panic.arg: (unreadable eval error: runtime.curg._panic is nil)
```

**Comment by jcburley** [comment]
Debugging this in **dlv** is proving challenging -- I prefer **gdb**, among other things.

**Comment by jcburley** [comment]
It's possible I've overlooked an important warning in the generated code, as I've generally paid little attention to those except when trying to track down a transpile or build error. So I'll focus on that next, reviewing the warnings in the **.go** code.

**Comment by jcburley** [comment]
Have been trying to remove more `typedef` usage, but sometimes **c2go** requires it in order to compile e.g. `map_node_t`, which is a self-referencing (via pointer) type, in order to generate correct code.

Having completed the performance analysis needed, this is now Deferred as a "nice to have" here, and I'll move on to the next task.

**Comment by jcburley** [comment]
The problem went away during iterative hand-conversion of the code to native Go facilities as of this commit:

https://github.com/jcburley/LispZeroGo/commit/0f1fe73aa75dece7352d856e8b4ef7d83cd61c4a

It's big mainly because of removal of c2go-generated `#include` headers that weren't needed. Restoring them temporarily had no effect on the generated code -- the tests still passed.

I suspect the `what` variable being changed from `*byte` to `string` was the culprit.

**Comment by jcburley** [comment]
The problem has since returned, but only when profiling is enabled (using either profiler currently supported).

I made a variety of changes, including converting list_read() to an iterative algorithm (to avoid running out of stack reading large inputs), but restoring the recursive algorithm doesn't avoid the crashes, so that isn't likely to be the problem.

Still feels like a GC problem to me -- and since I do want to understand some weird performance behavior I'm seeing (in particular, generating output, by not specifying the `-q` option to the program, takes vastly longer than does the corresponding C code, even though they both run about as fast with `-q`), I'll probably continue to eliminate `unsafe` and similar issues from the code to see if that gets rid of the crashes.

**Comment by jcburley** [comment]
One of the more-amusing errors that came out (only once though) was:

```
runtime: nelems=512 nalloc=92 previous allocCount=91 nfreed=65535
fatal error: sweep increased allocation count
```

For posterity, here's the complete output:

```
$ time LispZeroGo -cpuprofile ZERO-TEST zero-test.lisp 
runtime: cannot set cpu profile rate until previous profile has finished.
()
()
()
()
()
()
()
()
()
()
()
()
()
()
()
()
()
()
()
()
()
()
a
(d e f)
b
c
3
2
3
(4 should be 4)
(hey should be hey)
(t should be t)
(t should be t)
(() should be ())
(t should be t)
((a b c) should be (a b c))
(((d e f) (g h i)) should be ((d e f) (g h i)))
(((a b c) (g h i)) should be ((a b c) (g h i)))
(cond test : pass)
3
(x is i-am-x while y is i-am-y and yet mylist is ((a b c) (d e f) (g h i)))
()
()
()
()
bar
()
t
()
()
(a b)
()
()
runtime: nelems=512 nalloc=92 previous allocCount=91 nfreed=65535
fatal error: sweep increased allocation count

runtime stack:
runtime.throw(0x52e255, 0x20)
	/usr/local/go/src/runtime/panic.go:616 +0x81
runtime.(*mspan).sweep(0x7fe78b010988, 0x430f00, 0xc420000f00)
	/usr/local/go/src/runtime/mgcsweep.go:298 +0x89c
runtime.sweepone(0x431676)
	/usr/local/go/src/runtime/mgcsweep.go:113 +0x118
runtime.gosweepone.func1()
	/usr/local/go/src/runtime/mgcsweep.go:137 +0x2b
runtime.systemstack(0x7fff00000000)
	/usr/local/go/src/runtime/asm_amd64.s:409 +0x79
runtime.mstart()
	/usr/local/go/src/runtime/proc.go:1175

goroutine 3 [running]:
runtime.systemstack_switch()
	/usr/local/go/src/runtime/asm_amd64.s:363 fp=0xc420048f70 sp=0xc420048f68 pc=0x4551c0
runtime.gosweepone(0x0)
	/usr/local/go/src/runtime/mgcsweep.go:136 +0x4a fp=0xc420048fa0 sp=0xc420048f70 pc=0x41df1a
runtime.bgsweep(0xc420026070)
	/usr/local/go/src/runtime/mgcsweep.go:55 +0xb5 fp=0xc420048fd8 sp=0xc420048fa0 pc=0x41da75
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2361 +0x1 fp=0xc420048fe0 sp=0xc420048fd8 pc=0x457ce1
created by runtime.gcenable
	/usr/local/go/src/runtime/mgc.go:216 +0x58

goroutine 1 [runnable]:
main.atomp(...)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:133
main.listp(0xc4204a20c0, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:150 +0x1d7
main.list_car(0xc4204a20c0, 0xc4204f18e8)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:179 +0x2b
main.binding_lookup(0xc42034a97a, 0x6, 0xc42009a7b0, 0xc4204a27d0, 0xc4204f1a48)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:308 +0x562
main.binding_for(0xc42034a97a, 0x6, 0xc42009a7b0, 0xc4204a27d0, 0x1)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:547 +0x85
main.eval(0xc42034a97a, 0x6, 0xc42009bdd0, 0xc4204a27d0, 0x1)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:563 +0x852
main.eval(0xc42034a97a, 0x6, 0xc42038fcb0, 0xc4204a27d0, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:567 +0x44f
main.apply(0xc42034a97a, 0x6, 0xc42038fe30, 0xc42038fe30, 0xc420374cf0, 0xc4204a21f0, 0xc4204f21b5)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:652 +0x3b6
main.eval(0xc42034a97a, 0x6, 0xc420374ca0, 0xc4204a21f0, 0xc420490c30)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:577 +0x799
main.f_cons(0x528817, 0x4, 0xc420374c80, 0xc4204a21f0, 0xc42009a530)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:718 +0x13f
main.eval(0x528817, 0x4, 0xc420374be0, 0xc4204a21f0, 0xc420374b80)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.f_cond(0x528813, 0x4, 0xc420374d40, 0xc4204a21f0, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:769 +0x401
main.f_cond(0x528813, 0x4, 0xc420374b70, 0xc4204a21f0, 0xc42009a620)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:771 +0x464
main.eval(0x528813, 0x4, 0xc420374ad0, 0xc4204a21f0, 0xc420374ad0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.f_eval(0x528823, 0x4, 0xc42009a990, 0xc4204a21d0, 0xc42009a670)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:821 +0x383
main.eval(0x528823, 0x4, 0xc42009a8c0, 0xc4204a21d0, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.apply(0xc4200b0762, 0x6, 0xc42009a8a0, 0xc42009a8a0, 0xc42038f990, 0xc4204a2170, 0xc4204f2fed)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:652 +0x3b6
main.eval(0xc4200b0762, 0x6, 0xc42038f9a0, 0xc4204a2170, 0xc42038f9a0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:577 +0x799
main.f_eval(0x528823, 0x4, 0xc42009a990, 0xc4204a2150, 0xc42009a670)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:821 +0x383
main.eval(0x528823, 0x4, 0xc42009a8c0, 0xc4204a2150, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.apply(0xc4200b0762, 0x6, 0xc42009a8a0, 0xc42009a8a0, 0xc42038fca0, 0xc4204a20f0, 0xc4204f368d)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:652 +0x3b6
main.eval(0xc4200b0762, 0x6, 0xc42038fcb0, 0xc4204a20f0, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:577 +0x799
main.apply(0xc42034a97a, 0x6, 0xc42038fe30, 0xc42038fe30, 0xc42034c430, 0xc4204a2840, 0xc4204f399d)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:652 +0x3b6
main.eval(0xc42034a97a, 0x6, 0xc42034c3e0, 0xc4204a2840, 0xc42034c3e0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:577 +0x799
main.f_eval(0x528823, 0x4, 0xc420334440, 0xc4204a2030, 0xc42009a670)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:821 +0x383
main.eval(0x528823, 0x4, 0xc420334450, 0xc4204a2030, 0xc420334450)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.f_eval(0x528823, 0x4, 0xc42009abd0, 0xc4204a2090, 0xc42009a670)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:821 +0x383
main.eval(0x528823, 0x4, 0xc42009aaf0, 0xc4204a2090, 0xc4203244a0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.f_cons(0x528817, 0x4, 0xc42009aad0, 0xc4204a2090, 0xc42009a530)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:718 +0x13f
main.eval(0x528817, 0x4, 0xc42009a9d0, 0xc4204a2090, 0x1000001dc)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.f_cons(0x528817, 0x4, 0xc42009ac10, 0xc4204a2090, 0xc42009a530)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:717 +0xed
main.eval(0x528817, 0x4, 0xc42009a9b0, 0xc4204a2090, 0xc4204f4828)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.f_eval.func1(0xc42009a990, 0xc4204a2090, 0x528823, 0x4, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:818 +0x1a7
main.f_eval(0x528823, 0x4, 0xc42009a990, 0xc4204a2090, 0xc42009a670)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:820 +0x2fb
main.eval(0x528823, 0x4, 0xc42009a8c0, 0xc4204a2090, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.apply(0xc4200b0762, 0x6, 0xc42009a8a0, 0xc42009a8a0, 0xc420335560, 0xc4204a2030, 0xc4204f4da5)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:652 +0x3b6
main.eval(0xc4200b0762, 0x6, 0xc420335570, 0xc4204a2030, 0xc420335570)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:577 +0x799
main.f_eval(0x528823, 0x4, 0xc42009a990, 0xc4204a2db0, 0xc42009a670)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:821 +0x383
main.eval(0x528823, 0x4, 0xc42009a8c0, 0xc4204a2db0, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.apply(0xc4200b0762, 0x6, 0xc42009a8a0, 0xc42009a8a0, 0xc420335870, 0xc4204a2d50, 0xc420447445)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:652 +0x3b6
main.eval(0xc4200b0762, 0x6, 0xc420335880, 0xc4204a2d50, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:577 +0x799
main.apply(0xc4203261b8, 0x5, 0xc420335a00, 0xc420335a00, 0xc42034c3b0, 0xc4204a2840, 0xc420447755)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:652 +0x3b6
main.eval(0xc4203261b8, 0x5, 0xc42034c360, 0xc4204a2840, 0xc42034c360)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:577 +0x799
main.f_eval(0x528823, 0x4, 0xc42026ba20, 0xc4204a2c90, 0xc42009a670)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:821 +0x383
main.eval(0x528823, 0x4, 0xc42026ba30, 0xc4204a2c90, 0xc42026ba30)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.f_eval(0x528823, 0x4, 0xc42009abd0, 0xc4204a2cf0, 0xc42009a670)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:821 +0x383
main.eval(0x528823, 0x4, 0xc42009aaf0, 0xc4204a2cf0, 0xc420265e10)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.f_cons(0x528817, 0x4, 0xc42009aad0, 0xc4204a2cf0, 0xc42009a530)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:718 +0x13f
main.eval(0x528817, 0x4, 0xc42009a9d0, 0xc4204a2cf0, 0x1000001dc)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.f_cons(0x528817, 0x4, 0xc42009ac10, 0xc4204a2cf0, 0xc42009a530)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:717 +0xed
main.eval(0x528817, 0x4, 0xc42009a9b0, 0xc4204a2cf0, 0xc4204485e0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.f_eval.func1(0xc42009a990, 0xc4204a2cf0, 0x528823, 0x4, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:818 +0x1a7
main.f_eval(0x528823, 0x4, 0xc42009a990, 0xc4204a2cf0, 0xc42009a670)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:820 +0x2fb
main.eval(0x528823, 0x4, 0xc42009a8c0, 0xc4204a2cf0, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.apply(0xc4200b0762, 0x6, 0xc42009a8a0, 0xc42009a8a0, 0xc420304fd0, 0xc4204a2c90, 0xc420448b5d)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:652 +0x3b6
main.eval(0xc4200b0762, 0x6, 0xc420304fe0, 0xc4204a2c90, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:577 +0x799
main.apply(0xc420266d58, 0x7, 0xc420305160, 0xc420305160, 0xc42034c470, 0xc4204a2840, 0xc420448e6d)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:652 +0x3b6
main.eval(0xc420266d58, 0x7, 0xc42034c340, 0xc4204a2840, 0xc42034c340)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:577 +0x799
main.f_eval(0x528823, 0x4, 0xc420357ff0, 0xc4204a2bd0, 0xc42009a670)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:821 +0x383
main.eval(0x528823, 0x4, 0xc42035e000, 0xc4204a2bd0, 0xc42035e000)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.f_eval(0x528823, 0x4, 0xc42009abd0, 0xc4204a2c30, 0xc42009a670)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:821 +0x383
main.eval(0x528823, 0x4, 0xc42009aaf0, 0xc4204a2c30, 0xc420348eb0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.f_cons(0x528817, 0x4, 0xc42009aad0, 0xc4204a2c30, 0xc42009a530)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:718 +0x13f
main.eval(0x528817, 0x4, 0xc42009a9d0, 0xc4204a2c30, 0x1000001dc)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.f_cons(0x528817, 0x4, 0xc42009ac10, 0xc4204a2c30, 0xc42009a530)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:717 +0xed
main.eval(0x528817, 0x4, 0xc42009a9b0, 0xc4204a2c30, 0xc420449cf8)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.f_eval.func1(0xc42009a990, 0xc4204a2c30, 0x528823, 0x4, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:818 +0x1a7
main.f_eval(0x528823, 0x4, 0xc42009a990, 0xc4204a2c30, 0xc42009a670)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:820 +0x2fb
main.eval(0x528823, 0x4, 0xc42009a8c0, 0xc4204a2c30, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.apply(0xc4200b0762, 0x6, 0xc42009a8a0, 0xc42009a8a0, 0xc42035f110, 0xc4204a2bd0, 0xc42044a275)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:652 +0x3b6
main.eval(0xc4200b0762, 0x6, 0xc42035f120, 0xc4204a2bd0, 0xc42035f120)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:577 +0x799
main.f_eval(0x528823, 0x4, 0xc42009a990, 0xc4204a2ad0, 0xc42009a670)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:821 +0x383
main.eval(0x528823, 0x4, 0xc42009a8c0, 0xc4204a2ad0, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.apply(0xc4200b0762, 0x6, 0xc42009a8a0, 0xc42009a8a0, 0xc42035f420, 0xc4204a2a70, 0xc42044a915)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:652 +0x3b6
main.eval(0xc4200b0762, 0x6, 0xc42035f430, 0xc4204a2a70, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:577 +0x799
main.apply(0xc42033f8f8, 0x5, 0xc42035f5b0, 0xc42035f5b0, 0xc42034c320, 0xc4204a2840, 0xc42044ac25)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:652 +0x3b6
main.eval(0xc42033f8f8, 0x5, 0xc42034c2d0, 0xc4204a2840, 0xc4204a2a10)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:577 +0x799
main.f_cond(0x528813, 0x4, 0xc42034c4c0, 0xc4204a2840, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:769 +0x401
main.f_cond(0x528813, 0x4, 0xc42034c1d0, 0xc4204a2840, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:771 +0x464
main.f_cond(0x528813, 0x4, 0xc420349ec0, 0xc4204a2840, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:771 +0x464
main.f_cond(0x528813, 0x4, 0xc420348fc0, 0xc4204a2840, 0xc42009a620)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:771 +0x464
main.eval(0x528813, 0x4, 0xc420348ef0, 0xc4204a2840, 0xc420348ef0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.f_eval(0x528823, 0x4, 0xc42009a990, 0xc4204a2820, 0xc42009a670)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:821 +0x383
main.eval(0x528823, 0x4, 0xc42009a8c0, 0xc4204a2820, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.apply(0xc4200b0762, 0x6, 0xc42009a8a0, 0xc42009a8a0, 0xc42035f110, 0xc4204a27c0, 0xc42044b9dd)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:652 +0x3b6
main.eval(0xc4200b0762, 0x6, 0xc42035f120, 0xc4204a27c0, 0xc42035f120)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:577 +0x799
main.f_eval(0x528823, 0x4, 0xc42009a990, 0xc420495da0, 0xc42009a670)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:821 +0x383
main.eval(0x528823, 0x4, 0xc42009a8c0, 0xc420495da0, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:575 +0x732
main.apply(0xc4200b0762, 0x6, 0xc42009a8a0, 0xc42009a8a0, 0xc42035f420, 0xc420495d40, 0xc42044c07d)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:652 +0x3b6
main.eval(0xc4200b0762, 0x6, 0xc42035f430, 0xc420495d40, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:577 +0x799
main.apply(0xc42033f8f8, 0x5, 0xc42035f5b0, 0xc42035f5b0, 0xc420349e60, 0xc420495cd0, 0xc42044c38d)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:652 +0x3b6
main.eval(0xc42033f8f8, 0x5, 0xc420349d40, 0xc420495cd0, 0xc420349ce0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:577 +0x799
main.f_cond(0x528813, 0x4, 0xc420349ea0, 0xc420495cd0, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:769 +0x401
main.f_cond(0x528813, 0x4, 0xc420349cd0, 0xc420495cd0, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:771 +0x464
main.f_cond(0x528813, 0x4, 0xc420349b40, 0xc420495cd0, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:771 +0x464
main.f_cond(0x528813, 0x4, 0xc4203498f0, 0xc420495cd0, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:771 +0x464
main.f_cond(0x528813, 0x4, 0xc420349740, 0xc420495cd0, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:771 +0x464
main.f_cond(0x528813, 0x4, 0xc420349590, 0xc420495cd0, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:771 +0x464
main.f_cond(0x528813, 0x4, 0xc420349340, 0xc420495cd0, 0x0)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:771 +0x464
main.f_cond(0x528813, 0x4, 0xc420349190, 0xc420495cd0, 0xc42009a620)
	/home/craig/.go/src/github.com/jcburley/LispZeroGo/LispZeroGo.go:771 +0x464

goroutine 19 [syscall]:
os/signal.signal_recv(0x0)
	/usr/local/go/src/runtime/sigqueue.go:139 +0xa6
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x41

goroutine 20 [sleep]:
time.Sleep(0x5f5e100)
	/usr/local/go/src/runtime/time.go:102 +0x166
runtime/pprof.profileWriter(0x541060, 0xc4200b4020)
	/usr/local/go/src/runtime/pprof/pprof.go:750 +0x6d
created by runtime/pprof.StartCPUProfile
	/usr/local/go/src/runtime/pprof/pprof.go:735 +0x10e

real	0m31.445s
user	0m31.468s
sys	0m0.024s
```
